spring-cloud-starter-stream-source-sftp/src/main/java/org/springframework/cloud/stream/app/sftp/source/batch/SftpSourceBatchProperties.java

```diff
@@ -0,0 +1,185 @@
+
+	/**
+	 * Comma delimited list of environment properties to be applied to the
+	 * TaskLaunchRequest.
+	 */
+	private String environmentProperties;
+
+	/**
+	 * Value to use as the remote file job parameter name. Defaults to "remoteFilePath".
+	 */
+	private String remoteFilePathJobParameterName = DEFAULT_REMOTE_FILE_PATH_JOB_PARAM_NAME;
+
+	/**
+	 * Value to use as the local file job parameter name. Defaults to "localFilePath".
+	 */
+	private String localFilePathJobParameterName = DEFAULT_LOCAL_FILE_PATH_JOB_PARAM_NAME;
+
+	/**
+	 * The file path to use as the local file job parameter value. Defaults to "java.io.tmpdir".
+	 */
+	private String localFilePathJobParameterValue = new File(System.getProperty("java.io.tmpdir")).getAbsolutePath();
+
+	/**
+	 * Comma separated list of optional job parameters in key=value format.
+	 */
+	private List<String> jobParameters = new ArrayList<>();
+
+	@NotNull
+	public String getBatchResourceUri() {
+		return batchResourceUri;
+	}
+
+	public void setBatchResourceUri(String batchResourceUri) {
+		this.batchResourceUri = batchResourceUri;
+	}
+
+	@NotBlank
+	public String getDataSourceUrl() {
+		return dataSourceUrl;
+	}
+
+	public void setDataSourceUrl(String dataSourceUrl) {
+		this.dataSourceUrl = dataSourceUrl;
+	}
+
+	@NotBlank
+	public String getDataSourceUserName() {
+		return dataSourceUserName;
+	}
+
+	public void setDataSourceUserName(String dataSourceUserName) {
+		this.dataSourceUserName = dataSourceUserName;
+	}
+
+	public String getDataSourcePassword() {
+		return dataSourcePassword;
+	}
+
+	public void setDataSourcePassword(String dataSourcePassword) {
+		this.dataSourcePassword = dataSourcePassword;
+	}
+
+	public String getDeploymentProperties() {
+		return deploymentProperties;
+	}
+
+	public void setDeploymentProperties(String deploymentProperties) {
+		this.deploymentProperties = deploymentProperties;
+	}
+
+	public String getEnvironmentProperties() {
+		return environmentProperties;
+	}
+
+	public void setEnvironmentProperties(String environmentProperties) {
+		this.environmentProperties = environmentProperties;
+	}
+
+	public String getRemoteFilePathJobParameterName() {
+		if (remoteFilePathJobParameterName != null) {
+			return remoteFilePathJobParameterName.trim();
+		}
+
+		return null;
+	}
+
+	public void setRemoteFilePathJobParameterName(String remoteFilePathJobParameterName) {
+		this.remoteFilePathJobParameterName = remoteFilePathJobParameterName;
+	}
+
+	public String getLocalFilePathJobParameterName() {
+		if (localFilePathJobParameterName != null) {
+			return localFilePathJobParameterName.trim();
+		}
+
+		return null;
+	}
+
+	public void setLocalFilePathJobParameterName(String localFilePathJobParameterName) {
+		this.localFilePathJobParameterName = localFilePathJobParameterName;
+	}
+
+	@NotBlank
+	public String getLocalFilePathJobParameterValue() {
+		if (localFilePathJobParameterValue != null) {
+			return localFilePathJobParameterValue;
```

Comment:
This `if` sounds like redundant here -you are returning `null` otherwise anyway.
OR maybe `trim()` is lost...

Although from here I wonder why none of String-based configuration property doesn't have such a `trim()` option.
Maybe it is done automatically by the properties binder in Spring Boot? Or even earlier when config entries are read from the target file?

Anyway would be great to hear from you the reason for such an overhead with `trim()`